pennylane/ops/qubit/hamiltonian.py

```diff
@@ -619,18 +622,19 @@ def __iadd__(self, H):
         if isinstance(H, Hamiltonian):
             self._coeffs = qml.math.concatenate([self._coeffs, H.coeffs], axis=0)
             self._ops.extend(H.ops.copy())
-            self.simplify()
-            return self
 
-        if isinstance(H, (Tensor, Observable)):
+        elif isinstance(H, (Tensor, Observable)):
             self._coeffs = qml.math.concatenate(
                 [self._coeffs, qml.math.cast_like([1.0], self._coeffs)], axis=0
             )
             self._ops.append(H)
-            self.simplify()
-            return self
 
-        raise ValueError(f"Cannot add Hamiltonian and {type(H)}")
+        else:
+            raise ValueError(f"Cannot add Hamiltonian and {type(H)}")
+
+        self._wires = qml.wires.Wires.all_wires([op.wires for op in self.ops], sort=True)
```

Comment:
Warning: more than one target found for cross-reference 'matrix': pennylane.matrix, pennylane.ops.functions.matrix